> src/k8s.io/legacy-cloud-providers/... ./staging/src/k8s.io/metrics/... ./staging/src/k8s.io/mount-utils/... ./staging/src/k8s.io/pod-security-admission/... ./staging/src/k8s.io/sample-apiserver/... ./staging/src/k8s.io/sample-cli-plugin/... ./staging/src/k8s.io/sample-controller/...
ERROR: test/e2e/framework/util.go:620:12: appendAssign: append result not assigned to the same slice (gocritic)
ERROR: 	cmd.Env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_JOURNAL=%v", TestContext.DumpSystemdJournal))
ERROR: 	        

well the linter is not happy about the change

try:
env = os.Env...
env = append(env, .... )
env = append(env, .... )
cmd.Env = env

--- a/test/e2e/framework/util.go
+++ b/test/e2e/framework/util.go
@@ -617,7 +617,8 @@ func CoreDump(dir string) {
 	}
 	env := os.Environ()
 	env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_SERVICES=%s", parseSystemdServices(TestContext.SystemdServices)))
-	cmd.Env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_JOURNAL=%v", TestContext.DumpSystemdJournal))
+	env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_JOURNAL=%v", TestContext.DumpSystemdJournal))
+	cmd.Env = env
 
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr

--- a/test/e2e/framework/util.go
+++ b/test/e2e/framework/util.go
@@ -615,8 +615,9 @@ func CoreDump(dir string) {
 		Logf("Dumping logs locally to: %s", dir)
 		cmd = exec.Command(path.Join(TestContext.RepoRoot, "cluster", "log-dump", "log-dump.sh"), dir)
 	}
-	cmd.Env = append(os.Environ(), fmt.Sprintf("LOG_DUMP_SYSTEMD_SERVICES=%s", parseSystemdServices(TestContext.SystemdServices)))
-	cmd.Env = append(os.Environ(), fmt.Sprintf("LOG_DUMP_SYSTEMD_JOURNAL=%v", TestContext.DumpSystemdJournal))
+	env := os.Environ()
+	env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_SERVICES=%s", parseSystemdServices(TestContext.SystemdServices)))
+	cmd.Env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_JOURNAL=%v", TestContext.DumpSystemdJournal))
 
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr

